src/core_functions/scalar/list/list_sort.cpp

```diff
@@ -239,7 +258,9 @@ static unique_ptr<FunctionData> ListSortBind(ClientContext &context, ScalarFunct
 	bound_function.arguments[0] = arguments[0]->return_type;
 	bound_function.return_type = arguments[0]->return_type;
 	auto child_type = ListType::GetChildType(arguments[0]->return_type);
-
+	if (std::regex_search(bound_function.name, std::regex("grade_up"))) {
+		bound_function.return_type = LogicalType::LIST(LogicalTypeId::UINTEGER);
+	}
```

Comment:
See my previous comment about binding. You'll be in a custom bind function, so you don't have to add any if-else logic.